galsim/chromatic.py

```diff
@@ -19,45 +19,89 @@
-    GSObject works, except that it has access to the ChromaticObject methods described below;
-    especially expand(), dilate() and shift().
-
-    @param gsobj  The GSObject to be chromaticized.
+    A ChromaticObject can be instantiated from an existing GSObject by applying a
+    wavelength-dependent transformation.  E.g.,
+
+    >>> gsobj = galsim.Gaussian(fwhm=1)
+    >>> chrom_obj = gsobj.dilate(lambda wave: 1.01**wave)
+
+    In this and similar cases, the univariate object passed to the transformation method (`dilate`
+    above) is a python callable that accepts wavelength in nanometers and returns whatever type the
+    transformation method normally accepts (so an int or float above).
+
+    Another way to instantiate a ChromaticObject from a GSObject is to multiply by an SED.  This can
+    be useful to consistently generate the same galaxy observed through different filters, or, with
+    ChromaticSum, to construct multi-component galaxies, each component with a different SED.  For
+    example, a bulge+disk galaxy could be constructed:
+
+    >>> bulge_SED = user_function_to_get_bulge_spectrum()
+    >>> disk_SED = user_function_to_get_disk_spectrum()
+    >>> bulge_mono = galsim.DeVaucouleurs(half_light_radius=1.0)
+    >>> disk_mono = galsim.Exponential(half_light_radius=2.0)
+    >>> bulge = bulge_mono * bulge_SED
+    >>> disk = disk_mono * disk_SED
+    >>> gal = bulge + disk
+
+    The SEDs above describe the flux density in photons/nm/cm^2/s of an object, possibly normalized
+    with either the sed.withFlux(bandpass) or sed.withMagnitude(bandpass) methods (see the
+    docstrings in the SED class for details about these and other normalization options).  Note that
+    the `flux` attribute of the multiplied GSObject is accounted for in the newly constructed
+    ChromaticObject's normalization.  I.e., the following are equivalent:
+
+    >>> chrom_obj = (sed * 3.0) * gsobj
+    >>> chrom_obj2 = sed * (gsobj * 3.0)
+
+    Subclasses that instantiate a ChromaticObject directly also exist, such as ChromaticAtmosphere.
+    Even in this case, however, the underlying implementation always eventually wraps one or more
+    GSObjects.
+
+    ChromaticObjects can generally be sorted into two distinct types: those that represent galaxies
+    or stars and have dimensions of [photons/wavelength-interval/area/time/solid-angle], and those
+    that represent other types of wavelength dependence besides flux, like chromatic PSFs (these
+    have dimensions of [1/solid-angle]).  The former category of ChromaticObjects will have an SED
+    attribute that itself has its `spectral` attribute set to True, while the latter category of
+    ChromaticObjects will have an SED with its `dimensionless` attribute set to True.  These two
+    classes of ChromaticObjects have different restrictions associated with them.  For example, only
+    ChromaticObjects with a spectral SED can be drawn using chrom_obj.drawImage(bandpass, ...).
```

Comment:
In light of these restrictions, I was wondering if it might be worthwhile to add one more example to demo12?  Something that has a PSF with simple chromatic dependence that isn't being returned from a function that handles chromaticity under the hood?